servant-client/src/Servant/Common/Req.hs

```diff
@@ -148,7 +148,11 @@ performRequest reqMethod req manager reqHost = do
   partialRequest <- liftIO $ reqToRequest req reqHost
 
   let request = partialRequest { Client.method = reqMethod
+#if MIN_VERSION_http_client(0,5,0)
+                               , checkResponse = \ _req _res -> return ()
+#else
                                , checkStatus = \ _status _headers _cookies -> Nothing
+#endif
```

Comment:
I think it would be nicer to read if the CPP were used to implement two different variants of a top-level declaration. Something like `disableStatusCheck :: Response -> Response`.